File: src/main/java/bwdm/boundaryValueAnalysisUnit/BvaUnitMain.java
package bwdm.boundaryValueAnalysisUnit;

import bwdm.informationStore.InformationExtractor;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Objects;

public class BvaUnitMain {

	private InformationExtractor ie;
	private BoundaryValueAnalyzer boundaryValueAnalyzer;
	private ExpectedOutputDataGenerator expectedOutputDataGenerator;

	public BoundaryValueAnalyzer getBoundaryValueAnalyzer() { return boundaryValueAnalyzer; }

	public BvaUnitMain(InformationExtractor _ie) {
		this.ie = _ie;
		boundaryValueAnalyzer = new BoundaryValueAnalyzer(_ie);
		expectedOutputDataGenerator =
				new ExpectedOutputDataGenerator(
						_ie,
						Objects.requireNonNull(_ie.getIfElseExprSyntaxTree()).getRoot(),
						boundaryValueAnalyzer.getInputDataList()
				);

	}

	public String getAllTestcasesByBv() {
		StringBuilder buf = new StringBuilder();
		ArrayList<String> parameters = ie.getParameters();
		ArrayList<HashMap<String, Long>> inputDataList = boundaryValueAnalyzer.getInputDataList();
		ArrayList<String> expectedOutputDataList = expectedOutputDataGenerator.getExpectedOutputDataList();


		for(int i=0; i<expectedOutputDataList.size(); i++) {
			buf.append("No.").append(i + 1).append(" : ");
			HashMap<String, Long> inputData = inputDataList.get(i);
			for(String prm : parameters) {
				buf.append(inputData.get(prm)).append(" ");
			}
			buf.append("-> ").append(expectedOutputDataList.get(i)).append("\n");
		}

		return buf.toString();
	}



}

File: src/main/kotlin/bwdm/boundaryValueAnalysisUnit/BvaUnitMain.kt
package bwdm.boundaryValueAnalysisUnit

import bwdm.informationStore.IfElseExprSyntaxTree
import bwdm.informationStore.InformationExtractor

import java.util.Objects

class BvaUnitMain(private val ie: InformationExtractor) {
    val boundaryValueAnalyzer: BoundaryValueAnalyzer = BoundaryValueAnalyzer(ie)
    private val expectedOutputDataGenerator: ExpectedOutputDataGenerator

    val allTestcasesByBv: String
        get() {
            val buf = StringBuilder()
            val parameters = ie.parameters
            val inputDataList = boundaryValueAnalyzer.inputDataList
            val expectedOutputDataList = expectedOutputDataGenerator.expectedOutputDataList


            for (i in expectedOutputDataList.indices) {
                buf.append("No.").append(i + 1).append(" : ")
                val inputData = inputDataList[i]
                for (prm in parameters) {
                    buf.append(inputData[prm]).append(" ")
                }
                buf.append("-> ").append(expectedOutputDataList[i]).append("\n")
            }

            return buf.toString()
        }

    init {
        expectedOutputDataGenerator = ExpectedOutputDataGenerator(
                ie,
                Objects.requireNonNull<IfElseExprSyntaxTree>(ie.ifElseExprSyntaxTree).root,
                boundaryValueAnalyzer.inputDataList
        )

    }


}
